python/python/lance/torch/kmeans.py

```diff
@@ -217,7 +222,7 @@ def _fit_once(
         float
             The total distance of the current centroids and the input data.
         """
-        total_dist = 0
+        total_dist = torch.tensor(0.0, device=self.device)
```

Comment:
nit: I thought host 0D tensor is automatically propagate in the args memory. Is this requireed?

Reply:
Got any citations? I'd buy it being true the other way around (if we were writing to a tensor every time using the same, unchanged, float variable), since it would be an obvious optimization, but I can't find anything claiming this way around is true.